### PR TITLE
Add ServerName into TLS Config

### PR DESCRIPTION
**Pull request by gregorygtseng**
This PR covers a use case for monitoring certain hosts that don't have a DNS name, but still serves a certificate with a known CommonName. This exposes a new field in scrape configuration

    tls_config:
          server_name: 'example.com'

I'm not sure if the tests are sufficient because the error is masked by the presence of the test root CA. Could also DRY it up...

@beorn7 @fabxc @juliusv

### CODE REVIEW

**Comment by beorn7** [comment]
Seems fine to me, but I'm not the domain expert about the TLS config. @brian-brazil might have an opinion here.

File: config/config.go [hunk 1]
```diff
@@ -354,6 +354,8 @@ type TLSConfig struct {
 	CertFile string `yaml:"cert_file,omitempty"`
 	// The client key file for the targets.
 	KeyFile string `yaml:"key_file,omitempty"`
+	// Used to verify the hostname for the targets.
+	ServerName string `yaml:"servername,omitempty"`
```
Comment (brian-brazil) [inline config/config.go]: This should be server_name for consistency with the rest

Comment (gregorygtseng) [inline config/config.go]: Thanks, fixed and rebased

**Comment by grobie** [comment]
:+1:

File: retrieval/target_test.go [hunk 2]
```diff
@@ -283,6 +283,66 @@ func TestNewHTTPClientCert(t *testing.T) {
...
+	)
+	server.TLS = newTLSConfig(t)
+	server.StartTLS()
+	defer server.Close()
+
+	cfg := &config.ScrapeConfig{
+		ScrapeTimeout: model.Duration(1 * time.Second),
+		TLSConfig: config.TLSConfig{
+			CAFile:     "testdata/ca.cer",
+			ServerName: "127.0.0.1",
```
Comment (brian-brazil) [inline retrieval/target_test.go]: This is the same as the address, so this isn't really testing it. Can you add a 2nd cert with a different name?

**Comment by fabxc** [comment]
generally :+1:

💯 for providing tests

File: retrieval/target_test.go (hunk 2 above)
Comment (gregorygtseng) [inline retrieval/target_test.go]: Updated.

**Comment by brian-brazil** [comment]
Thanks!

**Comment by gregorygtseng** [comment]
ahh, I realized I exported that constant :/ had a force push but you're too fast

**Pull request merged**